### DISCUSSION TITLE
adding option to highlight text and comment on it

### DISCUSSION THREAD

**Discussion by lukaszgiez** [squidfunk/mkdocs-material#3358] [Q&A]
Hello!
Is it possible to implement a solution where the user would highlight a portion of the text on a generated page and comment on it (not commenting directly in markdown files)? Something that is available in MS Word and Confluence?

I really like MKdocs-material, but also really need this functionality. By the way, I don't mean comments at the bottom of the body of the page: I know these solutions exist.

I'd really appreciate your suggestions!

**Comment by squidfunk** [comment]
I think [Hypothesis](https://web.hypothes.is/) might be what you're looking for. Prior discussion in #1145.

**Reply by lukaszgiez** [reply]
thank you for the reply!

**Reply by lukaszgiez** [reply]
I've read the prior discussion and understand the integration with Mkdocs stalled.
I was wondering if anyone here could help me integrate Hypthesis into my Mkdocs documentation?
What I am trying to do is deploy the documentation and allow users with a link to it to add annotations. It looks like Hypothesis could provide this functionality, but I am not clear how to edit Mkdocs code to integrate Hypothesis.
Another thing I am not clear on is whether it's possible to keep the annotations on my server. I realize this is the question for Hypothesis people, but perhaps someone here knows.
If the info I give is too vague then please let me know what I should specify. I am new to technical documentation, not the most tech-savvy person, but I am trying to learn fast.
I'd greatly appreciate any help!

**Reply by squidfunk** [reply]
What you're asking is not exclusive to Material for MkDocs. It's probably better to ask somebody of the Hypothesis team or post your question on StackOverflow.

**Reply by lukaszgiez** [reply]
ok, thank you. I will do that.

**Comment by lukaszgiez** [comment]
Hello,
Matt, of Hypothesis, has reached out to me, and I enclose his answer to my questions. Hopefully, it will help someone in a similar situation to mine.
Btw, at present, I am using the solution with embedding Hypothesis directly into html. Basically, this line goes into the markdown files of my documentation:
`<script src="https://hypothes.is/embed.js" async></script>`
I will be looking into running my own Hypothesis server, but for now, this is enough to test the functionality.
Fortunately signing up with Hypothesis is a breeze.
Below is Matt's answer.
Greetings, folks!



> 
> I'm not familiar with Material for MkDocs, but typically there are two options available for embedding Hypothesis in a webpage without requiring a user to use the extension or bookmarklet.
> 
> The first is using by using our Via proxy server. This usually works well for publicly available sites without dynamically generated content. You can read more about Via here: https://web.hypothes.is/help/what-is-the-via-proxy/
> 
> The other option is adding code to the HTML that will embed Hypothesis directly in the page itself. This option requires that you have access and are able to modify the HTML sent to the user's browser. You can read more embedding Hypothesis here: https://web.hypothes.is/help/embedding-hypothesis-in-websites-and-platforms/
> 
> If you want to host your own instance of Hypothesis to keep annotations within your own environment, this is also possible but far more technically involved. You would need run your own Hypothesis server and configure for your own authority and user management. Documentation for this is here: https://h.readthedocs.io/en/latest/developing/
> 
> Hopefully this enough to get you started! Please let me know if you have any questions.
> 
> -Matt
